### PR TITLE
feat: Enable Nvidia driver builds for v535

### PR DESCRIPTION
**Pull request by p5**
This PR makes the required changes to build the v535 NVIDIA drivers.  

This can be merged when the packages have been released.  You can see which repos each package is in here
https://rpmfind.net/linux/rpm2html/search.php?query=akmod-nvidia

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,18 +20,20 @@ jobs:
...
             major_version: 38
           - driver_version: 520
             major_version: 37
           - driver_version: 520
             major_version: 38
-          - driver_version: 470
+          - driver_version: 525
+            major_version: 37
+          - driver_version: 525
             major_version: 38
```
Comment (p5) [inline .github/workflows/build.yml:34]: My OCD needed to fix the ordering of these...  Both lists are now from oldest to newest :)

File: build.sh [hunk 2]
```diff
@@ -15,7 +18,7 @@ else
 fi
 
 rpm-ostree install \
-    akmod-${NVIDIA_PACKAGE_NAME}*:${NVIDIA_MAJOR_VERSION}.*.fc${RELEASE} \
+    akmod-${NVIDIA_PACKAGE_NAME}-${NVIDIA_MAJOR_VERSION}.*.fc${RELEASE} \
```
Comment (p5) [inline build.sh:21]: Will need to confirm these with the older packages.  Works for v530 and v535, but not checked with the others

File: install.sh [hunk 3]
```diff
@@ -3,6 +3,9 @@
 set -ouex pipefail
 
 sed -i 's@enabled=1@enabled=0@g' /etc/yum.repos.d/fedora-{cisco-openh264,modular,updates-modular}.repo
+# Enable the RPM Fusion testing repositories for NVIDIA 535
+# https://rpmfind.net/linux/rpm2html/search.php?query=akmod-nvidia
+sed -i 's@enabled=0@enabled=1@g' /etc/yum.repos.d/rpmfusion-{free,nonfree}-updates-testing.repo
```
Comment (p5) [inline install.sh:8]: Will remove these when the packages are in the release repo.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -20,18 +20,12 @@ jobs:
       fail-fast: false
       matrix:
         major_version: [37, 38]
-        driver_version: [520, 525, 530, 470]
+        driver_version: [470, 530, 535]
```
Comment (p5) [inline .github/workflows/build.yml:23]: Removes the v520 and v525 drivers since they are being excluded for both Fedora versions

**Comment by p5** [comment]
I don't want this merging until v535 has been released to the regular reops and we've removed the sed commands which enable testing repos

**Comment by p5** [comment]
Feel free to take over my PR when the repos update if I don't notice.  
All that should need doing is removing the two `sed` commands I added for the testing repos and merging.  

We might want to set `is_latest_driver: false` for v535 until we've loaded this on a real system to test.  I'm not sure.

**Comment by p5** [comment]
AFAICT, v535 has been released to the proper repos, so I have updated this PR and made it ready to merge.

Only issue is Fedora 37 builds are failing, but I don't think that's due to any of my changes

**Review by EyeCantCU — APPROVED** [review]
Everything here looks good to me now that 535 is in stable. The cleanup of older driver versions is also nice. Thank you!

**Comment by MiMillieuh** [comment]
> AFAICT, v535 has been released to the proper repos, so I have updated this PR and made it ready to merge.
> 
> Only issue is Fedora 37 builds are failing, but I don't think that's due to any of my changes

This is due to the lack of the 530 driver in the F37 repo.

F37 RPMfusion has a 520 base and the update keeps only the latest driver so that means that Nvidia driver versions available for F37 are 340, 390, 520, 535

On the other hand F38 can be build with the 530 driver cause that the base driver version. so version that are available for F38 are  340, 390, 530, 535

That situation is normal and will happen each Nvidia driver update. the same issue will occurs if you try to keep 535 version on F38 when a newer version will be released.

In short terms you need to drop 530 for fedora 37 it won't build any more so that's pointless.

See pkgs.org to look for the avalible packages by yourself : https://pkgs.org/download/akmod-nvidia

Also you can check on the F36 repo that has literally the same behavior

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -20,16 +20,8 @@ jobs:
...
         exclude:
-          - driver_version: 525
-            major_version: 37
-          - driver_version: 525
-            major_version: 38
-          - driver_version: 520
-            major_version: 37
-          - driver_version: 520
-            major_version: 38
           - driver_version: 470
```
Comment (EyeCantCU) [inline .github/workflows/build.yml:25]: Exclude builds of 37 with driver version 530 here
```
- driver_version: 530
  major_version: 37
```

Comment (MiMillieuh) [reply .github/workflows/build.yml:25]: If you still wants some legacy in your version you can keep the 520 driver on fedora 37 cause it's the base driver so it will always be possible to build it.

Comment (p5) [reply .github/workflows/build.yml:25]: Changes made in 182798f

I don't think we need to worry about enabling an older driver version in this PR - this was mainly just getting the new ones to build

Comment (MiMillieuh) [reply .github/workflows/build.yml:25]: Sure that's fine to keep the latest only.

I was just saying this so you know what's possible to do or not in the current situation it might help understanding for the future so that kind of issue won't happen any more.
